test/test_helper.rb

```diff
@@ -87,3 +87,15 @@ def rule_without_metadata(rule)
   rule.delete(:_metadata)
   rule
 end
+
+def retry_test(delay = 0.1, max_retries = 30)
+  (1...max_retries).each do |i|
+    begin
+      return yield
+    rescue Algolia::AlgoliaHttpError
+      sleep delay * i
+    end
+  end
+
+  raise StandardError, 'reached the maximum number of retries'
```

Comment:
Can we make a custom error for this instead of using a `StandardError`? Something like a `TooManyRetries` error for example.

Reply:
In this repo we only have two kind of errors, `AlgoliaError` and `AlgoliaHttpError`. I found that it belonged to none of them so I stuck with the standard error